### PR TITLE
doc: Adds documentation for `replication_specs.*.external_id` computed attribute for `mongodbatlas_advanced_cluster` 

### PR DESCRIPTION
**Pull request by maastha**
## Description

Adds documentation for `replication_specs.*.external_id` computed attribute for `mongodbatlas_advanced_cluster`  and updates `replication_specs.*.id` description in `mongodbatlas_advanced_cluster`.

Link to any related issue(s): [CLOUDP-273692](https://jira.mongodb.org/browse/CLOUDP-273692)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [X] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: docs/resources/advanced_cluster.md [hunk 1]
```diff
@@ -533,6 +533,7 @@ replication_specs {
 }
 ```
 
+* `external_id` - Unique 24-hexadecimal digit string that identifies the replication object for a shard in a Cluster. When using the new sahrding configuration, this corresponds to Shard ID displayed in the UI.
```
Comment (AgustinBettati) [inline docs/resources/advanced_cluster.md]:
```suggestion
* `external_id` - Unique 24-hexadecimal digit string that identifies the replication object for a shard in a Cluster. This value corresponds to Shard ID displayed in the UI. When using old sharding configuration (replication spec with `num_shards` greater than 1) this value is not populated.
```
Small rephrasing suggestion to clarify this always corresponds to Shard ID, but is not populated if using old sharding configuration.

File: docs/resources/cluster.md [hunk 2]
```diff
@@ -420,7 +420,7 @@ replication_specs {
 
 * `num_shards` - (Required) Number of shards up to 50 to deploy for a sharded cluster. The resource returns 1 to indicate a replica set and values of 2 and higher to indicate a sharded cluster. The returned value equals the number of shards in the cluster.
 If you are upgrading a replica set to a sharded cluster, you cannot increase the number of shards in the same update request. You should wait until after the cluster has completed upgrading to sharded and you have reconnected all application clients to the MongoDB router before adding additional shards. Otherwise, your data might become inconsistent once MongoDB Cloud begins distributing data across shards. To learn more, see [Convert a replica set to a sharded cluster documentation](https://www.mongodb.com/docs/atlas/scale-cluster/#convert-a-replica-set-to-a-sharded-cluster) and [Convert a replica set to a sharded cluster tutorial](https://www.mongodb.com/docs/upcoming/tutorial/convert-replica-set-to-replicated-shard-cluster).
-* `id` - (Optional) Unique identifer of the replication document for a zone in a Global Cluster.
```
Comment (AgustinBettati) [inline docs/resources/cluster.md:423]: Should we take the opportunity to also document `id` in advanced_cluster? It is deprecated but could be good to document for transparency and adding same mention as for cluster resource

**Review by AgustinBettati — COMMENTED** [review]
small suggestion, thanks for grabbing this

File: docs/resources/cluster.md [hunk 3]
```diff
@@ -420,7 +420,7 @@ replication_specs {
...
+* `id` - (Optional) Unique identifer of the replication document for a zone in a Global Cluster. This value corresponds to the legacy sharding schema (no independent shard scaling) and is different from the Shard ID you may see in Atlas UI.
```
Comment (erabil-mdb) [inline docs/resources/cluster.md]:
```suggestion
* `id` - (Optional) Unique identifer of the replication document for a zone in a Global Cluster. This value corresponds to the legacy sharding schema (no independent shard scaling) and is different from the Shard ID you may see in the Atlas UI.
```

**Review by erabil-mdb — APPROVED** [review]
LGTM + minor suggestion!

File: docs/resources/cluster.md (hunk 2 above)
Comment (maastha) [reply docs/resources/cluster.md:423]: added

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Comment by lantoli** [comment]
@maastha @AgustinBettati should we add deprecation info in changelog?

**Comment by AgustinBettati** [comment]
@lantoli advanced_cluster `replication_specs.id` attribute was deprecated in 1.18.0 release and mentioned in [changelog](https://github.com/mongodb/atlas-sdk-go/pull/456). Here we are including it in docs (was not present) just clarify any potential confusion with new documented attribute `replication_specs.external_id`.

**Pull request merged**